discovery-infra/test_infra/utils/utils.py

```diff
@@ -492,9 +492,10 @@ def get_local_assisted_service_url(namespace, service, deploy_target):
     else:
         # default deploy target is minikube
         log.info('Getting minikube %s URL in %s namespace', service, namespace)
-        url, _, _ = run_command(
-            f'minikube -n {namespace} service {service} --url'
-        )
+        ip, _, _ = run_command("minikube ip")
+        # resolve the service node port form the service internal port (Support multiple ports per service).
+        port, _, _ = run_command(f'kubectl get svc assisted-service -n {namespace} -o=jsonpath="{{.spec.ports[?(@.port==8090)].nodePort}}"')
```

Comment:
Why it's part of the code? Why not passing it as variables?

Reply:
because minikube is a special case in this function

Reply:
@YuviGold what do you think?

Reply:
Could we just get the service external IP? when we work with the tunnel, the cluster has a load balancer and the service should be exposed. This is how we work with assisted-service when waiting for it to be healthy
https://github.com/openshift/assisted-service/blob/4f07ab7ea86cdf723d04b9e8a70e30b8e0f627b4/tools/utils.py#L116-L140
This way we won't need to have the minikube dependency.

Reply:
We should push to localhost, pushing to localhost doesn't require ssl...

Reply:
Sorry, I don't understand your answer...

Reply:
This code trying to resolve the assisted-service ip from the  Load balancer service....
minikube tunnel is like a node port service (minikube is a single node cluster) but the port is different.
Here i'm resolving the external port from the service internal port which is 8090.
please see the table below:

NAMESPACE            NAME               TYPE           CLUSTER-IP       EXTERNAL-IP      PORT(S)                          AGE
assisted-installer   assisted-service   LoadBalancer   10.104.184.193   10.104.184.193   8090:31951/TCP,40000:32463/TCP   44h
assisted-installer   object-store       ClusterIP      10.103.27.13     <none>           9000/TCP                         44h
assisted-installer   ocp-metal-ui       LoadBalancer   10.99.62.228     10.99.62.228     80:30370/TCP                     44h
assisted-installer   postgres           LoadBalancer   10.106.135.23    10.106.135.23    5432:32057/TCP                   44h
default              kubernetes         ClusterIP      10.96.0.1        <none>           443/TCP                          44h
kube-system          kube-dns           ClusterIP      10.96.0.10       <none>           53/UDP,53/TCP,9153/TCP           44h
kube-system          registry           ClusterIP      10.107.138.33    <none>           80/TCP,443/TCP                   44h

Reply:
I agree. I just want you to replace the `minikube ip` :)  So the solution would work with other clusters as well.

Reply:
how can we get the external ip...

Reply:
I linked above how we do it already in assisted-service deployment scripts, when waiting for the service to be healthy. :)

Reply:
do you suggest to extract it from: "kubectl get nodes -o=jsonpath={.items[0].status.addresses[0].address}" this is not a general solution...